emcc.py

```diff
@@ -1492,7 +1492,7 @@ def check(input_file):
       if shared.Settings.USE_PTHREADS == 2:
         exit_with_error('USE_PTHREADS=2 is not longer supported')
       if shared.Settings.ALLOW_MEMORY_GROWTH:
-        logging.warning('USE_PTHREADS + ALLOW_MEMORY_GROWTH may run non-wasm code slowly, see https://github.com/WebAssembly/design/issues/1271')
+        diagnostics.warning('warn-pthreads-mem-growth', 'USE_PTHREADS + ALLOW_MEMORY_GROWTH may run non-wasm code slowly, see https://github.com/WebAssembly/design/issues/1271')
```

Comment:
You don't need the `warn-` prefix as its redundant.  The uppercase `-W` implicitly reads and warning, so for example we might have `-Wunused` rather than `-Wwarn-unused`